src/content/blog/fastify.mdx

```diff
@@ -0,0 +1,179 @@
+
+### Fastify
+
+We landed on [fastify][fastify] as the technology to build the
+future of Val Town's public API. Some of the strongest points for
+fastify were:
+
+- An extremely strong [ecosystem](https://fastify.dev/ecosystem/),
+  which includes a lot of high-quality, first-party plugins. We're
+  building a system in production, which means things like rate
+  limiting are going to be necessary, and it's amazing that
+  fastify's [rate limiter](https://github.com/fastify/fastify-rate-limit)
+  is well-architected.
+- Great, thorough [documentation](https://fastify.dev/docs/latest/Reference/Routes/),
+  and an active community that has been really helpful for support.
+- Solid support for [plugins and encapsulation](https://fastify.dev/docs/latest/Reference/Encapsulation/) that let us structure the application into sub-applications.
+- Support for [validation and serialization](https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/) with multiple validators and validation for all inputs and ouputs.
+- [Good support for async/await](https://fastify.dev/docs/latest/Reference/Routes/#async-await), like you might expect from a modern framework.
+- Off-the-shelf integrations for [opentelemetry](https://www.npmjs.com/package/@opentelemetry/instrumentation-fastify) and [Sentry](https://docs.sentry.io/platforms/javascript/guides/fastify/) mean
+  that we could have tracing and telemetry from day one.
+- Finally, [@fastify/express](https://github.com/fastify/fastify-express)
+  let us incrementally migrate from express to fastify.
+- We were already using the [light-my-request](https://github.com/fastify/light-my-request) module to test
+  our express server. light-my-request is written and maintained
+  by the fastify folks, and it is built into fastify.
+
+Of course, fastify is also very fast, but web framework overhead
+wasn't a problem with express and most likely won't be a problem with
+any framework we choose: most of our performance envelope has to
+do with operating [the Val Town runtime][runtimes] and interacting
+with the database.
+
+We considered a few other options, like [Hono](https://hono.dev/) and [Elysia](https://elysiajs.com/).
+Both are lovely projects - we use and recommend Hono a ton
+in Vals. Elysia has convenient typesafety and a lot of features,
+but its support for Node.js is very experimental. And Hono has
+a [first-party OpenAPI plugin](https://github.com/honojs/middleware/tree/main/packages/zod-openapi), but some of the ecosystem around fastify - for tracing,
+incremental migration from express, and plugins - was more built-out.
+
+For simpler projects and especially for situations where we want
+multi-platform compatibility so that code can run on Node.js, Deno,
+or Cloudflare Workers, we reach for Hono.
+
+### The port
+
+By using [@fastify/express](https://github.com/fastify/fastify-express)
+we were able to do this migration entirely incrementally: the first
```

Comment:
In some ways I feel like this is the most interesting part of the post and I wonder if we can get it moved up higher or even in the title, ie "Migrating from Express to Fastify, one route a time"